[Term Entry] Go Math Functions Mod() (#2903)
Codecademy/docs: content/go/concepts/math-functions/terms/mod/mod.md

---
Title: 'Mod()'
Description: 'Returns the floating-point remainder of dividing x by y.'
Subjects:
  - 'Code Foundations'
  - 'Computer Science'
Tags:
  - 'Arithmetic'
  - 'Functions'
  - 'Numbers'
CatalogContent:
  - 'learn-go'
  - 'paths/computer-science'
---

The **`Mod()`** function returns the floating-point remainder of dividing *x* by *y*. The result has the same sign as *x* and a magnitude less than the magnitude of *y*.

> The **`Mod()`** function operates exclusively with floating-point numbers.

Reviewer: boldface is only applied to the first instance of the subject/concept in the entry

> If the numbers involved in the operation are of integer type, the `%` modulus operator should be used instead. The `%` operator works with integers, while **`math.Mod()`** operates with floating-point numbers.

Reviewer: ```suggestion
If the numbers involved in the operation are of integer type, the `%` modulus operator should be used instead. The `%` operator works with integers, while `math.Mod()` operates with floating-point numbers.
```

## Syntax

```pseudo
math.Mod(x,y)
```

The **`Mod()`** function takes two arguments of type 'float64'.
### Some special cases:

> 1. If the argument has a *+Inf* value, the return value will be *0*:

Reviewer: generally in the description it may be useful to highlight a formula or some term and I would advise using italics. In the syntax section, and in general, when referring to something in code or a specific language term/pkg/etc use backticks

2. If the value of *x* is either *(±)Inf* or *NAN*, the return value will be *NAN*.

3. The return value is *NAN* if the value of the second argument is either *0* or *NAN*.

4. If *(±)Inf* is passed as the second argument, the return value is *x (first argument)*.

## Example

> The following example demonstrate how to use the **`Mod()`** function and print the result:

Reviewer: ```suggestion
The following example demonstrates how to use the `Mod()` function and print the result:
```

```go
package main

import(
  "fmt"
  "math"
)

func main() {
  a:= 23.50
  b:= 5.20

  modResult := math.Mod(a, b)
  fmt.Printf("The modulus of %.2f and %.2f = %.2f", a, b, modResult)
}
```

The output will be:

```shell
The modulus of 23.50 and 5.20 = 2.70
```

## Codebyte Example

> The provided example is executable and demonstrates the usage of the **`Mod()`** function.

Reviewer: ```suggestion
The provided example is executable and demonstrates special cases in the use of the `Mod()` function:
```

```codebyte/golang
package main

import (
  "fmt"
  "math"
  "reflect"
)

func main() {
  mod_number := math.Mod(15.0, 2.75)
  fmt.Println(mod_number)
  fmt.Print("The data type is: ", reflect.TypeOf(mod_number))

> # Exceptions Examples
>   # First Exception Example

Reviewer: ```suggestion
```

    x := math.Inf(1) // Positive infinity
    result := math.Mod(10, x)
    fmt.Println(result)
    # The output will be: 0

> 

Reviewer: delete all of the output comments. The output will be visible when the user runs the code

---
Title: 'Mod()'
Description: 'Returns the floating-point remainder of dividing "x" by "y".'
Subjects:
  - 'Code Foundations'
  - 'Computer Science'
Tags:
  - 'Arithmetic'
  - 'Functions'
  - 'Numbers'
CatalogContent:
  - 'learn-go'
  - 'paths/computer-science'
---

The **`Mod()`** function returns the floating-point remainder of dividing **x** by **y**. The result has the same sign as **x** and a magnitude less than the magnitude of **y**.

> Mod function operates exclusively with floating-point numbers.

Reviewer: ```suggestion
The `Mod()` function operates exclusively with floating-point numbers.

```

> If the numbers involved in the operation are of integer type, the **%** modulus operator should be used instead. The % operator works with integers, while math.Mod operates with floating-point numbers.

Reviewer: ```suggestion
If the numbers involved in the operation are of integer type, the `%` modulus operator should be used instead. The `%` operator works with integers, while `math.Mod()` operates with floating-point numbers.
```

## Syntax

```pseudo
math.Mod(x,y)
```

The `Mod()` function takes two arguments of type 'float64'.

<!---blank--->

> 

Reviewer: These empty lines can be removed

> ### Some special cases:

Reviewer: add an empty line after the header

> 1. If the argument has a **+Inf** value, the return value will be **0**:

Reviewer: ```suggestion
1. If the argument has a `+Inf` value, the return value will be `0`:
```

Reviewer: Each of these is unnecessarily long. For example, (for number 1) keep either line 33 or 35, not both. The general aim of docs content is to provide clear and very concise explanations.

Edit the other numbers accordingly.

 
   If the argument has a value of +Inf (positive infinity), the return value will be 0. This reflects the mathematical truth that dividing any number by positive infinity results in a remainder of 0.
2. If the value of **x** is either **(±)Inf** or **NAN**, the return value will be **NAN**.

    If the value of x is either (±)Inf or NaN (Not a Number), the return value will be NaN. This is because using infinite or NaN values to calculate the remainder is mathematically undefined and yields an indeterminate result.
3. The return value is **NAN** if the value of the second argument is either **0** or **NAN**.

    The return value is NaN if the value of the second argument is either 0 or NaN. Division by zero or NaN in the modulus operation is mathematically meaningless, resulting in a NaN (Not a Number) value.

4. If **(±)Inf** is passed as the second argument, the return value is **x (first argument)**.

    If (±)Inf is passed as the second argument, the return value is x (the first argument). This is because dividing any number by positive or negative infinity results in the number itself as the remainder.

## Examples

> The following example demonstrates how to use the `Mod()` function  and print the result:

Reviewer: ```suggestion
The following examples demonstrate how to use the `Mod()` function and print the result:
```

```go
package main

import(
  "fmt"
  "math"
)

func main() {
  a:= 23.5
  b:= 5.2

  modResult := math.Mod(a, b)

>   fmt.Printf("%.2f mod %.2f = %.2f, a,b, modResult")

Reviewer: ```suggestion
  fmt.Printf("The modulus of %.2f and %.2f = %.2f", a, b, modResult)
```

}
```

The output will be:

```shell

> 23.50 mod 5.20 = 2.70

Reviewer: update the output accordingly

```

> >**Note:**

Reviewer: This is already addressed on line 19. I would suggest deleting line 74

In this example, **math.Mod()** calculates the remainder of dividing **23.5** by **5.2**, resulting in **2.7**. It's essential to note that the math.Mod function operates exclusively with **floating-point** numbers. If the numbers involved in the operation are of integer type, the **%** modulus operator should be used instead. The **%** operator works with integers, while math.Mod operates with floating-point numbers.

> #### Exceptions Examples 1.

Reviewer: Addressing all of the exceptions one by one with examples is a bit excessive. I would suggest showing a basic example and perhaps grouping all of the exceptions in the codebyte
